### ISSUE TITLE
Release 4.3.4 - Release Candidate 1 - External integrations modules

### ISSUE THREAD

**Issue by CarlosRS9** [wazuh/wazuh#13685]
The following issue aims to perform all the required testing for the current release candidate to ensure the modules (AWS, GCP, Azure and DockerListener) works as expected, report the results, and open new issues for any encountered errors.

## Modules tests information

|||
|----------------------------------|------ |
| **Main release candidate issue** | #13669     |
| **Version**                      | 4.3.4 |
| **Release candidate #**          | RC7   |
| **Tag**                          | [v4.3.4-rc1](https://github.com/wazuh/wazuh/tree/v4.3.4-rc1) |
| **Previous modules tests issue** | -    |

## Test report procedure

All test results must have one the following statuses: 
|                                  |                                                        |
|-----------------|-------------------------------------------------------------------------|
| :green_circle:  | The test passed successfully. 										    |
| :yellow_circle: | The test passed but some enhancements need to be applied to the module. |
| :red_circle:    | The test failed.                                                        |
| :white_circle:  | The test does not apply.                                                |

Any failing test must be properly addressed with a new issue, detailing the error and the possible cause. It must be included in the `Fixes` section of the current release candidate main issue.

The resulting logs for the tests must be included in the status report so it can be used by the auditors to dig deeper into any possible failures and details.

## Conclusions

| **Module**      | **Status**  |**Notes**|
|-----------------|-------------|---------|
| AWS             | :green_circle: Review/QA |  |
| GCP             | :green_circle: Review/QA |  |
| Azure           | :green_circle: Review/QA |  |
| Docker Listener | :green_circle: Review/QA |  |

## Auditors validation
The definition of done for this one is the validation of the conclusions and the test results from all auditors.

All checks from below must be accepted in order to close this issue.
- [x] @davidjiglesias 
- [x] @CarlosRS9

**Comment by CarlosRS9** [comment]
# Azure test results

The following tests must be performed in sequential order. No data must be modified or removed unless it was specified for the test. 

The tests assume there is a valid authentication file located at `{wazuh_path}/wodles/azure/credentials`.

A test is considered valid if the following conditions are met:

- The expected result described in the test itself is met.
- The expected amount of events were sent to `analysisd`.
- The expected output was present in the `{wazuh_path}/logs/azure-logs.log` file.

## Tests details

### Test Azure using ossec.conf

Remove the `/var/ossec/wodles/azure/last_dates.json` file and run the module using the following configuration. Every Graph log available should be processed (6). The `azure-logs.log` output must indicate that every query provided was executed. No duplicate events should be present.

<details><summary>Configuration used</summary>

    <wodle name="azure-logs">
        <disabled>no</disabled>
        <interval>10m</interval>
        <run_on_start>yes</run_on_start>

        <graph>
            <auth_path>/var/ossec/wodles/azure/credentials</auth_path>
            <tenantdomain>wazuh.onmicrosoft.com</tenantdomain>
            <request>
                <query>auditLogs/directoryAudits</query>
            </request>
        </graph>

        <log_analytics>
            <auth_path>/var/ossec/wodles/azure/credentials-analytics</auth_path>
            <tenantdomain>wazuh.onmicrosoft.com</tenantdomain>

            <request>
                <tag>azure-activity</tag>
                <query>AzureActivity</query>
                <workspace>XXXXXXXXXXXXXXXXXXXXXXXXXXX</workspace>
                <time_offset>140d</time_offset>
            </request>
        </log_analytics>

        <storage>
            <auth_path>/var/ossec/wodles/azure/credentials-storage</auth_path>
            <tag>azure-activity</tag>

            <container name="frameworktestcontainer">
                <content_type>json_inline</content_type>
                <time_offset>50d</time_offset>
            </container>

            <container name="frameworktestcontainer"></container>
        </storage>
    </wodle>

</details>

<details><summary>azure-logs.log output</summary>

</details>


<details><summary>Ossec.log output</summary>


</details>


<details><summary>last_dates.json contents after the test</summary>


</details>

#### Conclusions

**Comment by CarlosRS9** [comment]
# DockerListener tests results

Manual testing was performed to ensure everything works after the change. The tests were carried out using both a Wazuh manager and a Wazuh agent both using the following configuration:
```
<wodle name="docker-listener">
    <interval>30s</interval>
    <attempts>5</attempts>
    <run_on_start>yes</run_on_start>
    <disabled>no</disabled>
</wodle>
```

The [logall_json option](https://documentation.wazuh.com/current/user-manual/reference/ossec-conf/global.html#logall-json) must be enabled in the `ossec.conf` file before starting the tests.

### Test docker listener using ossec.conf in a manager
Restart Wazuh and check ossec.log to verify the module is running


<details><summary>ossec.log output</summary>

</details>



#### Conclusions
